include/psa/crypto_sizes.h

```diff
@@ -971,7 +971,8 @@
  */
 #define PSA_CIPHER_ENCRYPT_OUTPUT_MAX_SIZE(input_length)                        \
     (PSA_ROUND_UP_TO_MULTIPLE(PSA_BLOCK_CIPHER_BLOCK_MAX_SIZE,                  \
-                              (input_length) + PSA_BLOCK_CIPHER_BLOCK_MAX_SIZE))
+                              (input_length) + 1) +                             \
```

Comment:
I should have noticed earlier, but in general `round_up(max(block), i)` is not always the same as `max(round_up(block, i))`. For example, imagine the two possible block sizes are 7 and 9, and input length is 8: the ciphertext would be longer (14) with the smaller block size than with the largest (9, fits in one block). Now in practice I think it works because our possible block size tend to be powers of 2, hence the largest is a multiple of the smaller ones, which prevents things like the example above.

However, I think I'd be more comfortable having a formula that's correct in full generality, and not just under a specific assumption that happens to be correct for our current set of ciphers. The obvious formula would be to get rid of `ROUND_UP` here and just use `input_length + PSA_BLOCK_CIPHER_BLOCK_MAX_SIZE`: we're going to add at most 1 block of padding.

Well, actually I'm undecided, because `ROUND_UP` gives a tighter bound in most cases... Let's take time to discuss this before making changes.

Note: there are several other occurrences of this pattern.

Reply:
Ok, after hearing Dave's opinion, I'm with with keeping this as it is, as long as we have tests that would catch the error if we ever started having weird block sizes. I left a note in #3258 to remember to add a test for that.

No change required here.